### ISSUE TITLE
Video play/pause/restart not working in Windows IE 

### ISSUE THREAD

**Issue by trouille** [zooniverse/Panoptes-Front-End#2938]
## Expected behavior
Video subject w/in PBBP should play as expected on Windows Internet Explorer

## Current behavior
If you try to skip to a later section of the video, it moves you back to wherever you were. If you try to go to a previous section of the video, it moves you back to wherever you were. It also seems a bit stilted in its playing in general. 

A video of the experience:
https://drive.google.com/file/d/0BzeL_TYT7IYPWHlqU2xJVzRyaHM/view?usp=sharing

- **Operating system:**
Windows

- **Browser:**
Internet Explorer 11

**Comment by chrislintott** [comment]
We normally support the last two versions of major browsers; there have been three versions of Edge since IE11 so I'm not keen on changing much to support this.

**Comment by chrislintott** [comment]
Mind you, traffic from IE is still larger than edge...

**Comment by eatyourgreens** [comment]
I've seen behaviour like this in Firefox, usually if you try to jump forwards before the file has fully downloaded. I'm not sure if that's what's going on here but should we prevent people from playing/classifying until the full video has downloaded? That could be an annoying wait.

**Comment by shaunanoordin** [comment]
Debug Report
* On IE11, adjusting the Scrubber does not affect the Video being played.
* Tested on Win7+IE11

Replication:
* On IE11, go to https://www.zooniverse.org/projects/zooniverse/arizona-batwatch/classify
* Press Play to View the video being classified until the 10 sec (50%) time mark, then Pause.
* Move the video progress bar (Scrubber) to the 2 sec (10%) time mark, then Play
* Expected: video should continue playing from the 2 sec time mark.
* Actual: video plays from the 10 sec time mark.

Suspected area of wonkiness:
* `app/components/video-player.cjsx`
```
  seekVideo: () ->
    player = @refs.videoPlayer
    scrubber = @refs.videoScrubber
    time = scrubber.value
    player.currentTime = time
```
* Suspicion is that either `seekVideo()` isn't being called properly on IE11, or `player.currentTime = time` isn't registering properly. Maybe the former?

Analysis notes:
* At different points of the replication steps, I tried running `document.getElementsByClassName("video-scrubber")[0].value + " vs " + document.getElementsByClassName("subject")[1].currentTime;` in the browser consoles.
* On Chrome, Firefox, etc, the values of the Scrubber and the Video will always be the same. (Expected)
* On IE11,
  * The values are the same when the user Plays the Video.
  * But the values are different when the user adjusts the Scrubber.
  * (So following the replication steps above, the console would spew out "2 vs 10" instead of the expected "10 vs 10")

Next Steps:
* Confirming the cause of wonkiness. Bets on `input.onchange` not working for dumb IE-related reasons.

**Comment by shaunanoordin** [comment]
Update

Q: Where does it break in IE11?
A: In `app/components/video-player.cjsx`, the line in `render()` that says...
`<input type="range" className="video-scrubber" ref="videoScrubber" min="0" step="any" `**`onChange={@seekVideo}`**` />`
...doesn't register `onChange` properly.

Q: Why?
A: Dunno, it's IE11.

Q: Is there a way to make it work in IE11?
A: Change...
```
componentDidMount: ->
    @refs.videoScrubber?.value = 0
```
...to...
```
componentDidMount: ->
    @refs.videoScrubber?.addEventListener("change", @seekVideo)
    @refs.videoScrubber?.value = 0
```
...or better yet...
```
componentDidMount: ->
    if userIsUsingIE11()
      @refs.videoScrubber?.addEventListener("change", @seekVideo)
    @refs.videoScrubber?.value = 0
```

Now we know _why_ it breaks, but my proposed solution seems a bit hacky in our React context. I'm open to suggestions for improvement.

**Comment by srallen** [comment]
@shaunanoordin We have done an IE check in other places... it's hacky, but as long as it's temporary for the support of IE for just a bit longer, I'm ok with it. Perhaps we should abstract out this check into a js file in the lib folder: 
https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/lib/tutorial.cjsx#L94
and here:
https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/components/subject-viewer.cjsx#L73

**Comment by eatyourgreens** [comment]
Thanks @shaunanoordin this is hugely helpful 👍 

I've opened #2950 to rewrite the video player in JavaScript. I'll look at how that event listener is bound. Weird that it isn't being picked up in IE.

**Comment by eatyourgreens** [comment]
Staged at https://update-video-player.pfe-preview.zooniverse.org/projects/zooniverse/arizona-batwatch/classify?env=production which is probably still broken in IE11.

**Comment by eatyourgreens** [comment]
Looks like this is a known bug in React (even in 15, from reading the comments) https://github.com/facebook/react/issues/3096

**Comment by eatyourgreens** [comment]
Fingers crossed, this is fixed by be3cdfb931662402bea9f4310a8a7f1d8ef1b895 which also sniffs for `window.ActiveXObject`.

**Comment by shaunanoordin** [comment]
I've followed up on the notes from @srallen and @eatyourgreens (thanks!), and here are some updates on detecting IE11 using JS-ES6:

`if (window.ActiveXObject) { /*do IE stuff*/ }` won't work as Microsoft has [hidden the property from the DOM](https://msdn.microsoft.com/library/dn423948(v=vs.85).aspx).

`if ("ActiveXObject" in window) { /*do IE stuff*/ }` will work, however.
